File: java/src/main/java/DagMWE.java
import dag.Dag;
import dag.Vertex;
import dag.WeightComparison;
import dag.WeightMethods;

public class DagMWE {
    static class IntWeightI implements WeightMethods<Integer> {
        @Override
        public Integer add(Integer a, Integer b) {
            return a + b;
        }

        @Override
        public WeightComparison compare(Integer a, Integer b) {
            if (a > b) return WeightComparison.GREATER_THAN;
            if (a < b) return WeightComparison.LESS_THAN;

            return WeightComparison.EQUAL;
        }
    }

    public static void main(String[] args) {
        Dag<Integer> dag = new Dag<>(new IntWeightI());

        Vertex<Integer> a = dag.addVertex((1));
        Vertex<Integer> b = dag.addVertex((2));
        Vertex<Integer> c = dag.addVertex((3));

        dag.addEdge(a, b, (2));
        dag.addEdge(b, c, (3));
        dag.addEdge(a, c, (10));

        Integer pathWeight = dag.weightOfLongestPath(a, c,
                (i) -> i, (i) -> i);
        System.out.println(pathWeight);
    }
}

File: Haskell/DAG_MWE.hs
import DAG ( addVertex, initGraph, addEdge, weightOfLongestPath )

main :: IO ()
main = let 
    d0 = initGraph       
    (a, d1) = addVertex d0 (1::Int)
    (b, d2) = addVertex d1 2
    (c, d3) = addVertex d2 3
    d4  = addEdge d3 (a, b, 2)
    d5  = addEdge d4 (b, c, 3)
    dag = addEdge d5 (a, c, 10)
    in
        print $ weightOfLongestPath dag a c id id